projects/js-packages/svelte-data-sync-client/src/Endpoint.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod';
+import { API, RequestMethods, RequestParams } from './API';
+
+/**
+ * Every SyncedStore option has its own API Endpoint.
+ */
+export class API_Endpoint< T extends RequestParams > {
+	public nonce = '';
+
+	private endpoint: string;
+
+	constructor( private api: API, private name: string, private schema: z.ZodSchema ) {
+		/*
+		 * Convert `widget_name` to `widget-name` to match the endpoint.
+		 */
+		this.endpoint = this.name.replace( '_', '-' );
+	}
+
+	public async validatedRequest( method: RequestMethods = 'GET', params?: T ): Promise< T > {
+		const request = this.api.request( this.endpoint, method, this.nonce, params );
+		return await request.then( data => {
+			return this.schema.parse( data );
+		} );
+	}
+
+	public async GET(): Promise< T > {
+		return await this.validatedRequest( 'GET' );
+	}
+
+	public POST = async ( params: T ): Promise< T > => {
+		return await this.validatedRequest( 'POST', params );
+	};
+
+	public DELETE = async () => {
```

Comment:
Yeah, I just haven't encountered a situation where I need PATCH/PUT yet, so I haven't added those. But you're right - this needs some consistency, I'll update so that we only have post, get, and delete.